src/common/config_parser_test.cpp

```diff
@@ -581,3 +581,26 @@ TEST_F(ConfigParserTests, ValidateServerConfig) {
 	EXPECT_THAT(ret.error().String(), testing::HasSubstr("ServerURL"));
 	EXPECT_THAT(ret.error().String(), testing::HasSubstr("Servers"));
 }
+
+TEST_F(ConfigParserTests, CaseInsensitive) {
+	ofstream os(test_config_fname);
+	os << R"({
+  "artifactverifykey": "ArtifactVerifyKey_value",
+  "deviceTypeFile": "DeviceTypeFile_value",
+  "SERVERURL": "ServerURL_value"
```

Comment:
Maybe we should have a separate test where you have the same key with different casing? I'm a little bit interested to see how it handles it, especially when compared to having two completely identical keys.

Reply:
It behaves the same with two identical keys than with same key with different casing: the entry in the map gets overridden and only the last one stands.

I'll add a test, thanks for the suggestion.